components/merkledb/src/proof_map_index/tests.rs

```diff
@@ -83,11 +85,46 @@ fn generate_random_data_keys<R: Rng>(
     (0..len).map(kv_generator).collect::<Vec<_>>()
 }
 
-#[test]
-fn test_map_methods() {
+macro_rules! make_tests {
+    ($name:ident { $( ($field:ident, $upper:ident)),* }) => {
+        paste::item! {
+                $(
+                    #[test]
+                    fn [<$name _ $field>]()  {
+                        $name::<$upper>();
+                    }
+                )*
+        }
+    }
+}
+
+make_tests! { test_map_methods { (raw, Raw), (hashed, Hashed) } }
+make_tests! { test_insert_trivial { (raw, Raw), (hashed, Hashed) } }
+make_tests! { test_insert_same_key { (raw, Raw), (hashed, Hashed) } }
+make_tests! { test_insert_simple { (raw, Raw), (hashed, Hashed) } }
+make_tests! { test_insert_reverse { (raw, Raw), (hashed, Hashed) } }
+make_tests! { test_remove_trivial { (raw, Raw), (hashed, Hashed) } }
+make_tests! { test_remove_simple { (raw, Raw), (hashed, Hashed) } }
+make_tests! { test_remove_reverse { (raw, Raw), (hashed, Hashed) } }
+make_tests! { test_clear { (raw, Raw), (hashed, Hashed) } }
+make_tests! { test_fuzz_insert { (raw, Raw), (hashed, Hashed) } }
+make_tests! { test_build_proof_in_empty_tree { (raw, Raw), (hashed, Hashed) } }
+make_tests! { test_build_multiproof_in_empty_tree { (raw, Raw), (hashed, Hashed) } }
+make_tests! { test_build_proof_in_single_node_tree { (raw, Raw), (hashed, Hashed) } }
+make_tests! { test_build_multiproof_in_single_node_tree { (raw, Raw), (hashed, Hashed) } }
+make_tests! { test_build_proof_in_complex_tree { (raw, Raw), (hashed, Hashed) } }
```

Comment:
This test should fail for hashed `[u8; 32]` because it is specifically written with the assumption that the index keys are not transformed. Test does not fail because `[u8; 32]` has [identity `ObjectHash` implementation](https://github.com/exonum/exonum/blob/7547e85fe077a548c73caef574bee43e379ccb13/components/merkledb/src/hash.rs#L220), so in fact all `Raw` / `Hashed` test couples test the exact same thing 🙃 The `ObjectHash` implementation seems bogus to me; suggest to remove it.